### PR TITLE
turn button/icon to functional

### PR DESCRIPTION
**Pull request by Leopoldthecoder**
Please makes sure these boxes are checked before submitting your PR, thank you!

* [x] Make sure you follow Element's [Contributing Guide](https://github.com/ElemeFE/element/blob/master/.github/CONTRIBUTING.md).
* [x] Rebase before creating a PR to keep commit history clear.
* [x] Add some descriptions and refer relative issues for you PR.

### CODE REVIEW

**Comment by Leopoldthecoder** [comment]
例子里似乎完全没用到 `<el-icon>` 组件？

**Comment by QingWei-Li** [comment]
是的没用到 顺手做的感觉没卵用

File: packages/icon/src/icon.js [hunk 1]
```diff
@@ -0,0 +1,19 @@
+export default {
+  name: 'ElIcon',
+
+  functional: true,
+
+  props: {
```
Comment (QingWei-Li) [inline packages/icon/src/icon.js]: 好像应该用 propData？

Comment (Leopoldthecoder) [inline packages/icon/src/icon.js]: image
用 propsData 直接报错啦

Comment (QingWei-Li) [inline packages/icon/src/icon.js]: 哦 好厉害没搞懂干嘛的

Comment (Leopoldthecoder) [inline packages/icon/src/icon.js]: 说是为了方便单元测试

**Pull request merged**